src/test/storage/materialize_test.cpp

```diff
@@ -0,0 +1,114 @@
+    materialize_nulls<T>(column, nulls);
+    return nulls;
+  }
+
+  std::shared_ptr<const Table> _data_table;
+  std::shared_ptr<const Table> _references_table;
+  std::shared_ptr<const Table> _data_table_with_nulls;
+};
+
+TEST_P(MaterializeTest, MaterializeIntData) {
+  EXPECT_EQ(std::vector<int32_t>({12345, 123}),
+            materialize_values_to_vector<int32_t>(*_data_table->get_chunk(ChunkID(0))->get_column(ColumnID(0))));
+  EXPECT_EQ(std::vector<int32_t>({1234}),
+            materialize_values_to_vector<int32_t>(*_data_table->get_chunk(ChunkID(1))->get_column(ColumnID(0))));
+}
+
+TEST_P(MaterializeTest, MaterializeIntReferences) {
+  EXPECT_EQ(std::vector<int32_t>({12345, 123}),
+            materialize_values_to_vector<int32_t>(*_references_table->get_chunk(ChunkID(0))->get_column(ColumnID(0))));
+  EXPECT_EQ(std::vector<int32_t>({1234}),
+            materialize_values_to_vector<int32_t>(*_references_table->get_chunk(ChunkID(1))->get_column(ColumnID(0))));
+}
+
+TEST_P(MaterializeTest, MaterializeFloatData) {
+  const auto floats_0 =
+      materialize_values_to_vector<float>(*_data_table->get_chunk(ChunkID(0))->get_column(ColumnID(1)));
+  ASSERT_EQ(floats_0.size(), 2u);
+  EXPECT_FLOAT_EQ(458.7, floats_0.at(0));
+  EXPECT_FLOAT_EQ(456.7f, floats_0.at(1));
+
+  const auto floats_1 =
+      materialize_values_to_vector<float>(*_data_table->get_chunk(ChunkID(1))->get_column(ColumnID(1)));
+  ASSERT_EQ(floats_1.size(), 1u);
+  EXPECT_FLOAT_EQ(457.7, floats_1.at(0));
+}
+
+TEST_P(MaterializeTest, MaterializeFloatReferences) {
+  const auto floats_0 =
+      materialize_values_to_vector<float>(*_references_table->get_chunk(ChunkID(0))->get_column(ColumnID(1)));
+  ASSERT_EQ(floats_0.size(), 2u);
+  EXPECT_FLOAT_EQ(458.7, floats_0.at(0));
+  EXPECT_FLOAT_EQ(456.7f, floats_0.at(1));
+
+  const auto floats_1 =
+      materialize_values_to_vector<float>(*_references_table->get_chunk(ChunkID(1))->get_column(ColumnID(1)));
+  ASSERT_EQ(floats_1.size(), 1u);
+  EXPECT_FLOAT_EQ(457.7, floats_1.at(0));
+}
+
+TEST_P(MaterializeTest, MaterializeValuesAndNulls) {
+  const auto values_and_nulls = materialize_values_and_nulls_to_vector<int32_t>(
+      *_data_table_with_nulls->get_chunk(ChunkID(1))->get_column(ColumnID(0)));
+
+  ASSERT_EQ(values_and_nulls.size(), 2u);
+  EXPECT_EQ(values_and_nulls[0].first, true);
+  EXPECT_EQ(values_and_nulls[1].first, false);
+  EXPECT_EQ(values_and_nulls[1].second, 1234);
+}
+
+TEST_P(MaterializeTest, MaterializeNulls) {
+  const auto nulls =
+      materialize_nulls_to_vector<float>(*_data_table_with_nulls->get_chunk(ChunkID(0))->get_column(ColumnID(1)));
+  auto expected = std::vector<bool>{false, true};
+
+  EXPECT_EQ(expected, nulls);
+}
+
+INSTANTIATE_TEST_CASE_P(MaterializeTestInstances, MaterializeTest,
+                        ::testing::ValuesIn(std::begin(all_column_encoding_specs),
+                                            std::end(all_column_encoding_specs)), ); // NOLINT
```

Comment:
Could you add `_` here after `std::end(all_column_encoding_specs)), `? This is what we do in other places.

Reply:
According to Ctrl+Shift+F we don't ;)

Reply:
Yes we do. 

https://github.com/hyrise/hyrise/blob/75d8d8d7e6c58791abb0f7c8e07a3207f76e99df/src/lib/all_type_variant.hpp#L62